core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -134,11 +134,18 @@ class BpmBillingProjectLifecycle(
             )
             _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
           } yield CreationStatuses.CreatingLandingZone).recoverWith { case t: Throwable =>
+            logger.error("Billing project creation failed, cleaning up landing zone")
             cleanupLandingZone(landingZone.getLandingZoneId, projectName, ctx) >> Future.failed(t)
           }
         }
         .recoverWith { case t: Throwable =>
-          cleanupBillingProfile(profileModel.getId, projectName, ctx) >> Future.failed(t)
+          logger.error("Billing project creation failed, cleaning up billing profile")
+          cleanupBillingProfile(profileModel.getId, projectName, ctx).recover { case t: Throwable =>
+            logger.warn(
```

Comment:
I did not want to add this recover code to `cleanupBillingProfile` because in the case of deleting a billing project, I want the exception to pass up to the orchestrator.